src/main/java/io/leangen/geantyref/VarMap.java

```diff
@@ -95,7 +102,18 @@ AnnotatedType map(AnnotatedType type, MappingMode mappingMode) {
             if (!map.containsKey(tv)) {
                 if (mappingMode.equals(MappingMode.ALLOW_INCOMPLETE)) {
                     AnnotatedTypeVariable variable = (AnnotatedTypeVariable) type;
-                    AnnotatedType[] bounds = map(variable.getAnnotatedBounds(), mappingMode);
+                    AnnotatedType[] bounds;
+                    if(!currentlyResolvingTypeVariable.add(tv)) {
+                        //Issue #27: We have already seen this TypeVariable during the current resolution, so we need to abort
+                        //Otherwise, we get a StackOverflowError
+                        bounds = new AnnotatedType[]{new AnnotatedTypeImpl(Object.class)};
```

Comment:
Shouldn't it be just `return type` at this point? If I'm getting this right, at this point we know the variable isn't resolvable, so we can leave it as-is, right? That way the resulting type remains recursive, which I think is correct here.
What do you think?

Reply:
Hmm, some annotation merging maybe has to happen first 🤔 I have to think/tinker with it a little more.

Reply:
I though, I just break the recursion here with Object, which is IMHO the "best" guess, if we have a cycle, and let the other code merge the annotations etc. as it would normally.